### PR TITLE
Switch to GH Actions

### LINKED ISSUE #90: Use Github Actions for CI
CI is busted -- who knows how it worked before (travis? netlify? I'm sure we could remember but I don't right now).

Since this is on github and we're using github pages, let's just use github actions.

### PR DESCRIPTION
**Pull request by slifty**
This PR replaces our Travis CI configuration with GH pages.

I *believe* the various "secrets" referenced by the `publish.yml` workflow just exist by default and do not need to be explicitly set, so this should simply work when merged.

I admit I have no idea how to test that without merging -- thus is the eternal quandary of PRs that modify CI.

Resolves #90

### CODE REVIEW

**Comment by louh** [comment]
> I admit I have no idea how to test that without merging

Create a test branch that changes the "on" directive to target the test branch, e.g.:
  
    on:
      push:
        branches: [ main, test_branch ]

If it passes, then by the transitive property this branch shall also pass.

**Comment by chriszs** [comment]
You could also add a `workflow_dispatch` [trigger](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/) to run it manually.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Build
+        run: gatsby build
+
+      - name: Verify build
+        run: ls -la public
+
+      - name: Deploy with gh-pages
+        run: |
+          git remote set-url origin https://git:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+          npm run deploy -- -u "github-actions-bot <support+actions@github.com>"
```
Comment (chriszs) [inline .github/workflows/publish.yml:35]: `deploy` appears to also build, so we may be building twice. If we do it using npm or yarn we may not need to globally install the Gatsby CLI.

Comment (chriszs) [reply .github/workflows/publish.yml:35]: If we use a [deploy action](https://github.com/BadIdeaFactory/reoyez/blob/100a327ad1faaedb12ab485ba7d8c9f264786d6c/.github/workflows/publish.yml#L29-L33), we may not need the deploy Node dep.

Comment (slifty) [reply .github/workflows/publish.yml]: good catch -- removed the build / verify build steps!

Comment (chriszs) [reply .github/workflows/publish.yml]: Looks good. Could also get rid of `Install Gatsby CLI`.

**Comment by slifty** [comment]
Ty for all the tips!  Chris I ended up trying out `act` and it revealed that the site doesn't build in Node 14.

Updating the PR to specify 10 for our action (and I'm adding 10 to our package)

**Comment by slifty** [comment]
Here goes nothing.........

**Pull request merged**